src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCITriggerService.java

```diff
@@ -228,7 +238,15 @@ private RepositoryInfo getRepositoryInfo(ProgrammingExerciseParticipation partic
         String[] auxiliaryRepositoryUris = auxiliaryRepositories.stream().map(AuxiliaryRepository::getRepositoryUri).toArray(String[]::new);
         String[] auxiliaryRepositoryCheckoutDirectories1 = auxiliaryRepositories.stream().map(AuxiliaryRepository::getCheckoutDirectory).toArray(String[]::new);
 
-        if (programmingExercise.getCheckoutSolutionRepository()) {
+        ProgrammingExerciseBuildConfig buildConfig;
+        if (Hibernate.isInitialized(programmingExercise.getBuildConfig())) {
+            buildConfig = programmingExercise.getBuildConfig();
+        }
+        else {
+            buildConfig = programmingExerciseBuildConfigRepository.findByProgrammingExerciseId(programmingExercise.getId()).orElseThrow();
```

Comment:
Can you check where the exercise is retrieved and if we can also directly load the configuration there?
If this is currently the best solution, please use the `ElseThrow` variant of this method from the repository and do not manually throw an exception here

Reply:
Same here, the method is used in many places, where the buildconfig isn't needed throughout the whole process. And we only fetch it if we didn't previously